### ISSUE TITLE
Provide a working rsocket-kotlin and Android example

### ISSUE THREAD

**Issue by yschimke** [rsocket/rsocket-kotlin#164]
Out of https://stackoverflow.com/questions/67837839/is-it-possible-to-use-rsocket-in-android-with-kotlin

We should provide a working example android test, similar to https://github.com/square/okhttp/tree/master/android-test

**Comment by clayly** [comment]
Can i help somehow with such example? Is it suitable for Android in any way?

**Comment by yschimke** [comment]
What are you thinking? Add a test that runs in CI as described above? 

Alternatively a relatively clean sample that hit demo.rsocket.io would be a good starting point.

**Comment by clayly** [comment]
> What are you thinking? Add a test that runs in CI as described above?
> 
> Alternatively a relatively clean sample that hit demo.rsocket.io would be a good starting point.

I am working on it, diving into the project. In there something special about contributing what i should know?

**Comment by yschimke** [comment]
No, put up something for discussion and it should flush out any concerns.  Thanks!

**Comment by LittleTrickster** [comment]
I did create a demo https://github.com/LittleTrickster/AndroidRSocketDemo (it's a little bloated for a demo) that works both as a server and a client but there is still https://github.com/rsocket/rsocket-kotlin/issues/148 issue that crashes the whole app when the client loses connection to the server. 
Used unsafe workaround with Thread.setDefaultUncaughtExceptionHandler to prevent the crash for now.

**Comment by clayly** [comment]
> I did create a demo https://github.com/LittleTrickster/AndroidRSocketDemo (it's a little bloated for a demo) that works both as a server and a client but there is still #148 issue that crashes the whole app when the client loses connection to the server.
> Used unsafe workaround with Thread.setDefaultUncaughtExceptionHandler to prevent the crash for now.

Have you tried to make it work with okHttp?

**Comment by LittleTrickster** [comment]
> 
> 
> > I did create a demo https://github.com/LittleTrickster/AndroidRSocketDemo (it's a little bloated for a demo) that works both as a server and a client but there is still #148 issue that crashes the whole app when the client loses connection to the server.
> > Used unsafe workaround with Thread.setDefaultUncaughtExceptionHandler to prevent the crash for now.
> 
> Have you tried to make it work with okHttp?

Yeah it works just add  `android:usesCleartextTraffic="true"` to manifest.

**Comment by nailshakurov** [comment]
@LittleTrickster Is the sample project up to date?

**Comment by LittleTrickster** [comment]
@nailshakurov It's not.
